### DISCUSSION TITLE
How do you ingest Cisco syslog messages with 2.4?

### DISCUSSION THREAD

**Discussion by TotieBash** [Security-Onion-Solutions/securityonion#12055] [2.4]
### Version

2.4.30

### Installation Method

Security Onion ISO image

### Description

configuration

### Installation Type

Standalone

### Location

airgap

### Hardware Specs

Meets minimum requirements

### CPU

8

### RAM

32

### Storage for /

200

### Storage for /nsm

200

### Network Traffic Collection

tap

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

In SO2.3 I use filebeat Cisco ios module to ingest logs on UDP:9002.  With SO2.4 I believe the old method is no longer the way forward and there is no more so-filebeat docker container.  I think I need to put custom Cisco parser here /opt/so/saltstack/local/salt/elasticsearch/files/ingest/ but is that something I build from scratch or Elastic has a custom repo where I can just download these? 
https://docs.securityonion.net/en/2.4/elasticsearch.html#parsing

Also, I am unclear on how I configure the device, do I need to configure Cisco IOS with "logging host SO-IP transport udp port 514" or is it like SO2.3 where I send it to a custom port?

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by robbiemarshall** [comment]
You would use an Elastic Agent integration, but as of right now we do not support the Cisco IOS integration.  However it is on the list to be added and might be ready for 2.4.40, but no guarantee.  Additional documentation about the integration installation and configuration process can be found here:  https://docs.securityonion.net/en/2.4/elastic-fleet.html#integrations

**Reply by TotieBash** [reply]
@robbiemarshall I installed 2.4.40 and I did see the Cisco IOS integration and got it configured as per you link.  However it seems like that is not enough to get working.  When I do "docker ps" I don't see any docker container listening on port 9002.  Same with "iptables --list -n" and "ss -tulpn".  I assume so-logstash container needs to be listening on the port.   I configured a cisco ios device to "logging host SO transport udp port 9002" and verified it with "tcpdump -i [SO-MGMT-int] -nnn -A host [cisco ip]" that logs are coming in the wire.

**Reply by robbiemarshall** [reply]
Did you allow the host and port in the SO firewall?  The following link is specifically for the pfsense integration, but the steps will be the same for the Cisco integration.  Just make sure you use the correct port.  

https://docs.securityonion.net/en/2.4/pfsense.html#elastic-integration-for-pfsense

**Reply by TotieBash** [reply]
I just did, I missed the last step of the firewall config.  I can see it in "iptables --list" but I still don't see it in "ss -tulpn" or any docker container that has the port 9002 open.

**Reply by TotieBash** [reply]
Where is "Save and deploy changes"? Maybe that is the magic button I am looking.
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/86748904/2371656f-e069-4cd7-9dde-016b7570d10b)

**Reply by robbiemarshall** [reply]
That's just the "synchronize grid" option at the top of SOC when you're in Configuration, as shown here:

![Screenshot 2024-01-18 135808](https://github.com/Security-Onion-Solutions/securityonion/assets/122027497/e7a07b63-b6de-4fd7-bf9d-cc67ef078d61)

**Reply by TotieBash** [reply]
I did that, I even rebooted.  I see logs coming in tcpdump.  I should see the port 9002 open in "ss -tulpn" and "docker ps" correct?

**Reply by TotieBash** [reply]
### Ok I figured it out...
On step 1 of the instruction it says "click desired policy", I used the existing "so-grid-nodes_general" policy since it already has an elastic agent installed on my standalone host that is associated with it.  Previously I was creating new policy so it wasn't working since I would need to deploy an elastic-agent associated with it.
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/86748904/a72a766c-7776-413a-93c6-ea7f7f2d63bc)

### my "so-grid-nodes_general" policy:
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/86748904/5cfcc7e9-d3f2-4cf9-b613-ef7a285967e6)


### ss -tulpn should look like this:
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/86748904/d8318345-6d25-4693-93f3-3d54c8d85349)

### iptables should look like this:
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/86748904/18e6cd01-1c65-4fa1-abd3-876efa2b0b6b)

### SO Kibana dashboard will have cisco.ios dataset:
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/86748904/36e6f0fc-cdd7-4611-97a9-e848427394db)
